### DISCUSSION TITLE
Performance improvements in version 2.x?

### DISCUSSION THREAD

**Discussion by pawelbaranski** [pdezwart/php-amqp#470] [Q&A] [answered]
Hi

Is there a performance improvement between versions 1 and 2 or is it mostly an improvement of an API and package design ?

**Comment by lstrojny** [comment, marked as answer]
The 2.0 release does not bring much in terms of performance improvements. There is not so terribly much PHP AMQP can actually do in terms of performance as it mostly hinges on how efficient librabbitmq is (hint: quite efficient) and how slow or fast it is to bridge into the PHP ecosystem (quite fast nowadays).

Upgrading is still very much worth it, if only for the numerous bug fixes. 

Are you looking for specific performance improvements, anything that is slow for you and is tied to the extension?

**Reply by pawelbaranski** [reply]
nothing in particular, I just didn't find any notes regarding that topic, that's why I asked. Thanks for the explanation
